src/ngamsCore/ngamsLib/ngamsDbCore.py

```diff
@@ -819,6 +822,10 @@ def _data_to_bind(self, data):
 
     def _prepare_query(self, sql, args):
 
+        if not self._use_prepared_statement:
+            markers = ["'{}'" if isinstance(arg, str) else "{}" for arg in args]
+            return sql.format(*markers).format(*args), ()
```

Comment:
I think this logic is a tad too simple to work correctly. It will work for parameters that are strings themselves, but, for example, not when the argument is a list of strings (is that even a possibility? I'm actually not sure). And it needs to escape single quotes in the strings too.

It'd be great to actually test this too. A suggestion would be to add another job to the CI job matrix where we locally modify the `test/src/ngamsCfg.xml` file to add `UsePreparedStatement="true"`, then let the tests run as usual.

Reply:
@rtobar good points!
* logic was improved to handle `None` values and single quotes
* I added a test to the matrix for `Python3.9`, `MySQL` and `UsePreparedStatement="false"`. Now all tests are passing!
* in my last commit I fixed exiting code not passing arguments to `cursor.execute(sql)` if the arguments are empty. This logic was already implemented in `transaction` but not in `ngamsDbCursor`